addons/mail/static/src/models/messaging_menu/messaging_menu.js

```diff
@@ -34,7 +34,14 @@ function factory(dependencies) {
         /**
          * Toggle whether the messaging menu is open or not.
          */
-        toggleOpen() {
+        async toggleOpen() {
+            if (!this.isOpen) {
+                const inbox = this.env.messaging.inbox;
+                if (!inbox.mainCache.isLoaded) {
+                    // populate some needaction messages on threads.
+                    await inbox.mainCache.loadMessages();
+                }
+            }
```

Comment:
This really needs to be changed. It is not acceptable to toggle in general, and especially here when doing it after an async rpc, all conditions are here to get inconsistencies.

Reply:
Ok, just add `!isLoading` in condition